File: library-frontend/src/types/library.ts
export interface Library {
  id: string;
  name: string;
  ownerId: string;
  ownerName: string;
  address: string | null;
  phone: string | null;
  active: boolean;
}

export interface CreateLibraryRequest {
  name: string;
  address?: string;
  phone?: string;
}

export interface UpdateLibraryRequest {
  name?: string;
  address?: string;
  phone?: string;
}

File: library/src/main/java/com/lab/library/entity/Library.java
package com.lab.library.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Table(name = "libraries")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Library extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private UUID ownerId;

    private String address;

    private String phone;

    @Column(nullable = false)
    private boolean active = true;
}
